### ISSUE TITLE
Rich text paste from any source needs to strip all font colors and font backgrounds

### ISSUE THREAD

**Issue by SigmasonicX** [OffprintStudios/dragonfish#267]
Text copied and pasted from Google Docs is explicitly black, which users often don't realize. This is accurately copying the color attributes of the copied text, but because this happens without users noticing much of the time, it isn't a desirable experience.

Either the code should detect if the copied text color is explicitly black and then remove the color attribute if so; or it should not copy over text color at all.

**Comment by Figments** [comment]
this comment was in #215 but will be posted here since it's a related issue:

> Reported by Krickis:
> So when copying stuff over from certain websites it gives this funky highlight, which can not be removed. This is what it looks like when it uses the remove font color and remove highlight options on it. This was copied from Fimfiction (it only does this if copying a single paragraph; if copying two or more paragraphs it copies without issue)
https://cdn.discordapp.com/attachments/690694506231627787/769041616240377866/Screenshot_176.png
> 
> I know how to fix it (paste as plain text, manually add the link back in) but it was previously mentioned that the remove highlight option would work on this and it does not
> 
> Issue can be seen in the author's note of this chapter
> 
> I suspect this is due to how FimFiction will highlight a paragraph when you click it and give you an option to quote it for a comment.

**Comment by pingzing** [comment]
For the sake of provoking some discussion, I'd like to propose an alternative. First, my reasoning:

Most authors aren't going to be using text colors at all in their stories. For those authors who do, it'll be be something that's probably only used in a few places. (Ignoring really genre-specific stuff, like a story trying to ape the "Homestuck feeling" for a moment). In this scenario, it's very easy to imagine an author spending a while getting their colors just right in their editor, then not even noticing that the Offprint editor has stripped it away until after publish time.
In addition, I don't think it's particularly urgent that we fix this right away--we're in alpha after all. I think it's more important that we fix this _right_, with good UX, and clear messaging to the user. Mostly because anything we do here is likely to be labor-intensive, and very fiddly to get right. I could be wrong! But I suspect I'm not.

Now, my proposal for fixing this has two parts:

## The "works for 90% of people" easy fix:

Yeah, strip all colors out. This is probably even desirable for a lot of people pasting from places like GDocs. However, I think we should also do two things:
1) Have a checkbox that indicates "strip colors" is enabled with a tooltip that explains what it does.
2) Flash a message on-screen if we had to strip colors upon paste to warn a user that we've modified their text, and to introduce them to the feature.

## The "I know what I'm doing, stop holding my hand":

This is what happens when that shiny checkbox is disabled. We stop stripping colors on paste. However, I _guarantee_ we're going to get some users who hate it when The Computer does things to their precious story text, and disable the strip-on-paste out of habit. They will then complain that their text colors are wrong. In order to reduce this, we _probably_ want to give users some way of previewing their text in arbitrary color themes.

tl;dr:

- Checkbox to disable/enable stripping colors on paste
- Checkbox defaults to _checked_
- Flash a warning when we strip colors to warn users that we Did A Thing
- Allow users to preview story content in arbitrary themes so that authors who need colors don't hate life

**Comment by Figments** [comment]
I agree with all of the above, tbh. we're gonna inevitably run into people who will demand our heads on pikes for stripping their text colors, so having color strip on by default with a preview so people who want their colors intact can preview in dark/light themes would solve the problem, I think

**Comment by SigmasonicX** [comment]
Status before investigating linked tickets:
Will be resolved by https://github.com/OffprintStudios/dragonfish/pull/437
Which is blocked by https://github.com/OffprintStudios/dragonfish/issues/436

**Comment by Figments** [comment]
fixed by migrating to tiptap
